lib/livebook/sessions.ex

```diff
@@ -18,8 +18,8 @@ defmodule Livebook.Sessions do
     opts = Keyword.put(opts, :id, id)
 
     case DynamicSupervisor.start_child(Livebook.SessionSupervisor, {Session, opts}) do
-      {:ok, pid} ->
-        {:ok, Session.get_by_pid(pid)}
+      {:ok, _pid, session} ->
+        {:ok, session}
```

Comment:
I'm pretty sure the problem is that in the test the session auto shutdowns between we start it and call `Session.get_by_pid`. To solve I changed the start_link to return the session as info.